_posts/2025-04-22-openloco-v25.04.md

```diff
@@ -0,0 +1,72 @@
+---
+title:      "OpenLoco version 25.04"
+tagline:    "Release blog post"
+author:     "Aaron van Geffen"
+date:       2025-04-22 19:40:00 +0200
+layout:     single
+classes:    wide
+categories: news
+---
+
+OpenLoco v25.04 is out! This month, we've made a bunch of progress on the reimplementation of the company AI,
+as well as a few exciting new features. Let's dig into the details in this dev log. For a summary of changes
+please find the [changelog](https://github.com/OpenLoco/OpenLoco/releases/tag/v25.04) on GitHub.
+
+## A faster way to build tracks
+
+Locomotion's UI represented quite a departure from the _Transport Tycoon_ games that preceded it.
+This is perhaps most notable in the way tracks and roads are built in the game, which clearly inherits
+from the _RollerCoaster Tycoon_ games Locomotion shares its engine with. While the resulting
+construction window provides a quick overview of available track pieces, it doesn't necessarily
+make for the fastest way to construct those tracks.
+
+In earlier versions of OpenLoco, we tried to mitigate this somewhat by allowing players to
+hold down the construction button for repeated building. This still only sped things up a little, though.
+This month, we have a more exciting change: you can now _drag_ the construction marker to
+fill an area with the selected track pieces ([#3073](https://github.com/OpenLoco/OpenLoco/pull/3073))!
+Here's quick video of the new tool in action:
+
+<video style="width: 90%; margin: auto" controls loop>
+	<source src="/assets/mp4/construction_drag.mp4" type="video/mp4">
+</video>
+
+To achieve this, the selection tool is now activated when the mouse button is _released_ (`ToolUp` event)
+instead of immediately when the button is _pressed_ (`ToolDown` event). Now, when `ToolDown` is activated,
+the initial tile position is stored. When the player starts moving the mouse (`ToolDrag` event),
+the _current_ tile position is stored. If the two positions aren't identical, the new 'dragging' mode is activated.
+You'll now just see a map selection grid, showing what tiles will be filled with track pieces.
+If you let you, the new pieces will be built immediately.
+
+Conversely, if the two positions are identical, then the game takes the vanilla code path:
+you'll see a ghost track piece, and clicking will build that piece, and activate 'interactive' mode,
+just like it always did.
+
+NB: this currently only extends to the track/road and station construction tabs. The signal and overhead tabs
+have not been adapted. We're not sure how best to implement them yet, but they will likely be adapted in the
+future as well.
+
+
+## More work on the company AI
+
+_TODO: write some words on Duncan's wizardry here_
```

Comment:
This month it was working out what vehicle types to use and building up a train from the vehicle types. Applying the playstyle restrictions (as mentioned in previous month). Choosing where to place stations and how big to make them. Replacing AI allocated track/road (the ghost you can't see that prevents the player building) with real track/road. And finally removeing all track / road e.g. when going bankrupt.